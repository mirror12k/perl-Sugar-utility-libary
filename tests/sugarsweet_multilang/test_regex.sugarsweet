
class TestClass {

	sub _constructor(tree args) {
		if ("%asdf%f" =~ /%(.*?)%/) {
			print join ': ', ["\t[TestClass] my match", match[1]];
		}

		if ("HELLOWORLD" =~ /owo/i) {
			print join ': ', ["\t[TestClass] my uwu match", match[0], "at", match.pos];
		}

		if ("HELLOWORLD" !~ /dwo/i) {
			print join ': ', ["\t[TestClass] cant find dwo"];
		}

		while ("asdf qwer zxcv whoami lol" =~ /(\w+)/g) {
			print join ': ', ["\t[TestClass] matched something: ", match[1]];
		}
	}

	# main
	void sub main () native:perl5 {{
		my $f = TestClass->new({
			message => 'hello from perl5!',
		});
	}}

	void sub main () native:php7 {{
		$f = new TestClass([
			'message' => 'hello from php7!',
		]);
	}}

	void sub main () native:python3 {{
		f = TestClass({
			'message':'hello from python3!',
		})
	}}

	void sub main () native:javascript6 {{
		var f = new TestClass({
			'message':'hello from javascript6!',
		})
	}}

}

