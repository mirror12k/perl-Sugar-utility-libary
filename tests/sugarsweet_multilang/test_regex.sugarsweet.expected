	[TestClass] my match: asdf
	[TestClass] my uwu match: OWO: at: 7
	[TestClass] cant find dwo
	[TestClass] matched something: : asdf
	[TestClass] matched something: : qwer
	[TestClass] matched something: : zxcv
	[TestClass] matched something: : whoami
	[TestClass] matched something: : lol
